packages/x-tree-view/src/TreeItem/TreeItem.tsx

```diff
@@ -329,12 +346,27 @@ export const TreeItem = React.forwardRef(function TreeItem(
     className: classes.dragAndDropOverlay,
   });
 
+  const ErrorContainer: React.ElementType = slots.errorContainer ?? TreeItemErrorContainer;
+  const errorContainerProps = useSlotProps({
+    elementType: ErrorContainer,
+    getSlotProps: getErrorContainerProps,
+    externalSlotProps: slotProps.errorContainer,
+    ownerState: {},
+    className: classes.errorContainer,
+  });
+
   return (
     <TreeItemProvider {...getContextProviderProps()}>
       <Root {...rootProps}>
         <Content {...contentProps}>
           <IconContainer {...iconContainerProps}>
-            <TreeItemIcon status={status} slots={slots} slotProps={slotProps} />
+            {status.error && <ErrorContainer {...errorContainerProps} />}
+
+            {status.loading ? (
+              <CircularProgress size="12px" sx={{ color: 'text.primary' }} thickness={6} />
```

Comment:
I suspect some people will want to customize it, but we can wait for feedback before making it a slot

Reply:
Considering I made the `ErrorContainer` a slot, so for consistency, I should probably make this one a slot as well 🤔